src/defaults.ts

```diff
@@ -55,16 +64,41 @@ export const DEFAULT_OPTIONS = {
    * @param baseURL The `baseURL` from `options.baseURL`
    * @returns {string} The returned string will become the `innerHTML` of the tooltip element
    */
-  formatTooltip: formatValue,
+  formatTooltip?: (value: any, valueToHtml: (value: any) => string, maxDepth: number, baseURL: string) => string;
 
   /**
    * The baseurl to use in image paths.
    */
+  baseURL?: string;
+
+  /**
+   * The snap reference for the tooltip.
+   */
+  anchor?: 'cursor' | 'mark';
+
+  /**
+   * The position of the tooltip relative to the anchor.
+   *
+   * Only valid when `anchor` is set to 'mark'.
+   */
+  position?: Position | Position[];
+}
+
+export const DEFAULT_OPTIONS: Required<Options> = {
+  offsetX: 10,
+  offsetY: 10,
+  id: EL_ID,
+  styleId: 'vega-tooltip-style',
+  theme: 'light',
+  disableDefaultStyle: false,
+  sanitize: escapeHTML,
+  maxDepth: 2,
+  formatTooltip: formatValue,
   baseURL: '',
+  anchor: 'cursor',
+  position: ['top', 'bottom', 'left', 'right', 'top-left', 'top-right', 'bottom-left', 'bottom-right'],
 };
 
-export type Options = Partial<typeof DEFAULT_OPTIONS>;
```

Comment:
What was wrong with this?

Reply:
There wasn't anything wrong previously but when adding `anchor` and `position`, the method didn't work well.

Originally the types were inferred from the default object. Since the types are inferred, any string default will be inferred as type `string`. This worked great for all the previous properties of `Options`.

`anchor` and `position` are both string unions. So there is a set list of acceptable strings that can be provided. If we let the types be inferred by the default object, then the type for these would be inferred as `string` instead of `cursor` | `mark` etc.

The advantage of these two using a string union is intelliSense and general type checking. If a consumer is using TS, then they will get an error if they set `anchor = 'mouse'`. And all consumer (TS of JS) should get intelliSense suggestions as long intelliSense is enabled.

<img width="209" alt="image" src="https://github.com/user-attachments/assets/dd8467e5-0ab1-4e1f-95f9-2cbb17f4e023">

This is why I flipped the order so we define the `Options` type first and then define the `DEFAULT_OPTIONS` and set it's type to `Options`.

Reply:
You can add `as const` or `as 'cursor' | 'mark` to the object, I think, to make the type stricter. But I don't feel strongly about flipping this so no need to change.